asciigraf/asciigraf.py

```diff
@@ -358,8 +358,11 @@ def node_iter(network_string):
             (Point(0,0), node1), (Point(9,0), (label1))
         )
     """
+    NODE_MATCH = re.compile(
+        r'([\w{}()\]\[]([\w:;,`\'\".]+[ ^ ]?)*)?[\w:;,`\'\".)?!.{}\[\]()]+'
```

Comment:
![](https://i.imgflip.com/4h10mp.jpg)

Reply:
Is there anyway we can write a regex that matches anything except the control characters? Might be more generic since this regex only works on ascii characters

Reply:
right, like basically saying anything not a dash that would be a line? i think that could work, the only issue i see there is it would make it harder to add edge modifiers, like for example if you wanted to do something like `n1<--------n2` for a directed graph.

Reply:
Good point on edge modifiers. Do we have an idea of how those might be supported or what we'd want to support?

Reply:
the simplest approach would be to use `<` `>` and `^` `v` i think. then you could have graphs like:
```
    n1------->n2
    ^              |    
    |              v
    n3<-------n4

```

I think it wouldn't really be crazy to just disallow the first three, but `v` seems like a problem -- we'd have to disallow the node `v`, which is a little counterintuitive and also would add a complicated edge-case to the node extraction logic. 


the other option is to do unicode arrows `→ ← ↓ ↑ ↖ ↙ ↘ ↗`.

```
← ↓ ↑ ↖ ↙ ↘ 
n1---------→ n2---n5
↑            |    /
|            ↓  ↙
n3←----------n4
```

Not sure i like the diagonal arrows, but the horizontal arrows are pretty clean, and it wouldn't cause problems with node labels.


In terms of *how* to support them i think the special characters all map onto existing edge characters so you just do the same thing with both '|' and `^` .


Ok but devils advocate here: this library is pretty stable, maybe it would make sense to just do a breaking change if the edge-modifier thing does happen? and just support any non-edge characters for now.

Reply:
I would avoid unicode arrows ... it'll just make it so much more complex.

Yeah, we can do a breaking change later if we need directional links. Right now what are the use cases for it?

Reply:
the use case i'm interested in is adding things to a node, like

```

------n1<----L1
      ^
      |
 L2--/

```

If asciigraf could parse this, i would collapse all the directional edges and use `L1` and `L2` to modify `n1` as node attributes -- without this feature, its not really straightforward to tell which edges are like this.

Reply:
ok, updated the regex to match any group that has no edge characters and doesn't have more than one white-space. also multi-lined it and put comments so it would be easier to read